substrate/frame/system/src/lib.rs

```diff
@@ -877,7 +885,17 @@ pub mod pallet {
 		///
 		/// All origins are allowed.
 		#[pallet::call_index(11)]
-		#[pallet::weight((T::SystemWeightInfo::apply_authorized_upgrade(), DispatchClass::Operational))]
+		#[pallet::weight((
+			T::SystemWeightInfo::apply_authorized_upgrade()
+				.saturating_add(T::SystemWeightInfo::validate_apply_authorized_upgrade()),
+			DispatchClass::Operational,
+		))]
+		#[pallet::authorize(|_source, code| Pallet::<T>::validate_apply_authorized_upgrade(code).map(|v| (v, Weight::zero())))]
+		// Because the weight of both "validate unsigned" and "authorize" logic needs to be added
+		// to the weight of the extrinsic, and because "validate unsigned" weight must be included
+		// in the call weight, the "authorize weight is also included in the call weight.
+		// Thus we set it to zero here.
```

Comment:
This comment could have a `TODO`, just to make it more clear on what needs to be done once `ValidateUnsigned` is removed.

```suggestion
		// Thus we set it to zero here.
		// Once `ValidateUnsigned` is deprecated and removed, `T::SystemWeightInfo::validate_apply_authorized_upgrade()` can be moved from `#[pallet::weight(...)]` to `#[pallet::weight_of_authorize(...)]` attribute.
```

Reply:
Not really important, just noticed that the actual_weight returned is `T::BlockWeights::get().max_block`.